includes/Rewrites.php

```diff
@@ -211,12 +211,24 @@ function store_template( $template ) {
      *
      * @return string
      */
-    function store_toc_template( $template ) {
+    public function store_toc_template( $template ) {
+        $store_name = get_query_var( $this->custom_store_url );
+
         if ( ! $this->is_woo_installed() ) {
             return $template;
         }
-        if ( get_query_var( 'toc' ) ) {
-            return dokan_locate_template( 'store-toc.php' );
+
+        if ( ! empty( $store_name ) ) {
```

Comment:
We should check the existence of query var `toc`, then we should fetch the seller. That's a performance improvement I guess. If the query var wasn't there, still the user will be fetched. So we can defer the fetching later once we know the query var we are expecting was the right one.

Reply:
Yes, that's a better idea. I will update the PR.